R/ExperimentHub-class.R

```diff
@@ -17,15 +17,15 @@ ExperimentHub <-
              ask=getExperimentHubOption("ASK"))
 {
     if (is.null(proxy)){
-        connect <- curl::has_internet()
+        connect <- !is.null(curl::nslookup("experimenthub.bioconductor.org", error=FALSE))
     } else {
         connect <- TRUE
-        message("Cannot determine internet connection.",
-                "\n If you experience connection issues consider ",
-                "using 'localHub=TRUE'")
+        message(sprintf("Assuming valid proxy connection through '%s'.", proxy),
```

Comment:
`sprintf()` add little here compared to the `paste0()` implicit in message / warning / stop
```
message(
    "Assuming valid proxy connection through '", proxy, "'",
    "\n  If you experience connection issues consider ",
    "using 'localHub=TRUE'"
)
```

Reply:
@LTLA I'm not convinced that changing from `has_internet` to `nslookup` is a good idea.  `has_internet` calls `nslookup` and does other checks regarding proxy.   What would be great is if you could put in a pull request to curl for `has_internet` to include an argument  say `nslookup_host` that could be optionally passed.  In the meantime I can change the messages to the suggested improvements.

Reply:
I don't mind making the PR but its use here would require the expected IP addresses are for ExperimentHub. `has_internet` compares the IP to `8.8.8.8` and `8.8.4.4`, corresponding to the Google DNS server; we'd need to replace these values for `has_internet` to work.

Reply:
I still think its better to do the pull request to curl  and see what they say.  Pointing out that certain locations block google but still have internet (I believe that is what you are saying is the issue?) --  I'm updating the messages manually to include in ExperimentHub as well as AnnotationHub.

Reply:
I suppose I should actually check that it solves the problem in LTLA/SingleR#109.

Reply:
Yes that would also be good.  Closing this pull request. Updating the messages manually.

Reply:
@LTLA  Did you confirm that it would fix the issue?  I will reconsider this if it is confirmed.